doc/api/buffer.md

```diff
@@ -2604,6 +2348,245 @@ in UTF-16 code units.
 
 This value may depend on the JS engine that is being used.
 
+## The `--zero-fill-buffers` command line option
+<!-- YAML
+added: v5.10.0
+-->
+Node.js can be started using the `--zero-fill-buffers` command line option to
+force all newly allocated `Buffer` instances created using either
+`new Buffer(size)`, [`Buffer.allocUnsafe()`], [`Buffer.allocUnsafeSlow()`] or
+`new SlowBuffer(size)` to be *automatically zero-filled* upon creation. Use of
+this flag *changes the default behavior* of these methods and *can have a significant
+impact* on performance. Use of the `--zero-fill-buffers` option is recommended
+only when necessary to enforce that newly allocated `Buffer` instances cannot
+contain potentially sensitive data.
+
+Example:
+
+```txt
+$ node --zero-fill-buffers
+> Buffer.allocUnsafe(5);
+<Buffer 00 00 00 00 00>
+```
+
+## Deprecated `new Buffer()` Constructors
+
+In versions of Node.js prior to 6.0.0, `Buffer` instances were created using the
+`new Buffer()` constructor function, which allocates the returned `Buffer`
+differently based on what arguments are provided:
+
+* Passing a number as the first argument to `Buffer()` (e.g. `new Buffer(10)`),
+  allocates a new `Buffer` object of the specified size. Prior to Node.js 8.0.0,
+  the memory allocated for such `Buffer` instances is *not* initialized and
+  *can contain sensitive data*. Such `Buffer` instances *must* be subsequently
+  initialized by using either [`buf.fill(0)`][`buf.fill()`] or by writing to the
+  `Buffer` completely. While this behavior is *intentional* to improve
+  performance, development experience has demonstrated that a more explicit
+  distinction is required between creating a fast-but-uninitialized `Buffer`
+  versus creating a slower-but-safer `Buffer`. Starting in Node.js 8.0.0,
+  `Buffer(num)` and `new Buffer(num)` will return a `Buffer` with initialized
+  memory.
+* Passing a string, array, or `Buffer` as the first argument copies the
+  passed object's data into the `Buffer`.
+* Passing an [`ArrayBuffer`] or a [`SharedArrayBuffer`] returns a `Buffer` that
+  shares allocated memory with the given array buffer.
+
+Because the behavior of `new Buffer()` changes significantly based on the type
+of value passed as the first argument, applications that do not properly
+validate the input arguments passed to `new Buffer()`, or that fail to
+appropriately initialize newly allocated `Buffer` content, can inadvertently
+introduce security and reliability issues into their code.
+
+To make the creation of `Buffer` instances more reliable and less error prone,
```

Comment:
I know this was in the existing text, but while we're in here, can you hyphenate `error-prone`?